tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -3298,6 +3243,76 @@ private:
       return cachedRDF;
    }
 
+   template <typename F>
+   RInterface<Proxied, DS_t>
+   VaryImpl(const std::vector<std::string> &colNames, F &&expression, const ColumnNames_t &inputColumns,
+            const std::vector<std::string> &variationTags, std::string_view variationName)
+   {
+      using F_t = std::decay_t<F>;
+      using ColTypes_t = typename TTraits::CallableTraits<F_t>::arg_types;
+      using RetType = typename TTraits::CallableTraits<F_t>::ret_type;
+      constexpr auto nColumns = ColTypes_t::list_size;
+
+      SanityChecksForVary<RetType>(colNames, variationTags, variationName);
```

Comment:
I try hard to make function bodies fit in a single screen.

`JittedVaryImpl` does similar checks to `SanityChecksForVary` inline -- all the ones it can do without compile-time column type information.